### PR TITLE
[WebSub] Basic auth support for WebSub Hub

### PR DESCRIPTION
**Pull request by chamil321**
> Before passing the hubListener, its configuration should be populated with AuthProvider as follows.
``` ballerina
http:AuthProvider basicAuthProvider = {
    scheme: "basic",
    authStoreProvider: "config"
};

http:ServiceEndpointConfiguration hubListenerConfig = {
    authProviders: [basicAuthProvider],
    secureSocket: {
        keyStore: {
            path: "${ballerina.home}/bre/security/ballerinaKeystore.p12",
            password: "ballerina"
        },
        trustStore: {
            path: "${ballerina.home}/bre/security/ballerinaTruststore.p12",
            password: "ballerina"
        }
    }
};
```
> Then this config is used to create and start the hub
```ballerina
var val = websub:startHub(new http:Listener (9191, config =  hubListenerConfig));
```
> In addition to authProviders, User needs to specify the authConfig properties at service config level. It is done via config API. Following values should be mentioned in the config file alone with user credentials

```
["b7a.websub.hub.auth"]
enabled=true  // enables the authentication
scopes="scope1" // defines the scope of possible users

["b7a.users"]

["b7a.users.tom"]
password="1234"
scopes="scope1"
```
> The subscriber(client) should define auth config in subscriptionClientConfig as follows
```ballerina
@websub:SubscriberServiceConfig {
    path: "/ordereventsubscriber",
    subscriptionClientConfig: {
        auth: {
            scheme: http:BASIC_AUTH,
            username: "tom",
            password: "1234"
        }
    }
}
```

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > added

### CODE REVIEW

File: tests/ballerina-integration-test/src/test/resources/websub/advanced_services/01_websub_publisher.bal [hunk 1]
```diff
@@ -23,6 +23,25 @@ import ballerina/websub;
...
+            path: "${ballerina.home}/bre/security/ballerinaKeystore.p12",
+            password: "ballerina"
+        },
+        trustStore: {
+            path: "${ballerina.home}/bre/security/ballerinaTruststore.p12",
+            password: "ballerina"
+        }
+    }
+};
+
```
Comment (MaryamZi) [inline tests/ballerina-integration-test/src/test/resources/websub/advanced_services/01_websub_publisher.bal]: We should also add a test for publishing via the hub client when auth is enabled right? Shall we create an issue, if not possible with this PR?

Comment (chamil321) [reply tests/ballerina-integration-test/src/test/resources/websub/advanced_services/01_websub_publisher.bal]: Added a test case

File: tests/ballerina-integration-test-utils/src/main/java/org/ballerinalang/test/context/BServerInstance.java [hunk 2]
```diff
@@ -354,9 +369,10 @@ private void startServer(String[] args, Map<String, String> envProperties,
             process = processBuilder.start();
 
             serverInfoLogReader = new ServerLogReader("inputStream", process.getInputStream());
-            tmpLeechers.forEach(leacher -> serverInfoLogReader.addLeecher(leacher));
+            tmpInfoLeechers.forEach(leacher -> serverInfoLogReader.addLeecher(leacher));
             serverInfoLogReader.start();
             serverErrorLogReader = new ServerLogReader("errorStream", process.getErrorStream());
+            tmpErrorLeechers.forEach(leacher -> serverErrorLogReader.addLeecher(leacher));
```
Comment (MaryamZi) [inline tests/ballerina-integration-test-utils/src/main/java/org/ballerinalang/test/context/BServerInstance.java]: "leacher" --> "leecher"

File: tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/service/websub/advanced/WebSubSecureHubTestCase.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    private static final String INTENT_VERIFICATION_SUBSCRIBER_THREE_LOG =
+            "Subscription Request failed at Hub[https://localhost:9191/websub/hub], for Topic[http://one.persistence" +
+                    ".topic.com]: Error in request: Mode[subscribe] at Hub[https://localhost:9191/websub/hub] - " +
+                    "Authorization failure ";
+    private static final String INTENT_VERIFICATION_SUBSCRIBER_FOUR_LOG = "ballerina: Intent Verification agreed - " +
+            "Mode [subscribe], Topic [http://one.websub.topic.com], Lease Seconds [1200]";
+    private static final String INTERNAL_HUB_NOTIFICATION_SUBSCRIBER_ONE_LOG =
+            "WebSub Notification Received by One: {\"mode\":\"internal\", \"content_type\":\"json\"}";
+    private static final String INTERNAL_HUB_NOTIFICATION_SUBSCRIBER_TWO_LOG =
+            "WebSub Notification Received by Four: {\"mode\":\"remote\", \"content_type\":\"xml\"}";
```
Comment (MaryamZi) [inline tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/service/websub/advanced/WebSubSecureHubTestCase.java]: should we maybe test this based on a different payload? 

`{\"mode\":\"remote\", \"content_type\":\"xml\"}` was originally meant to remotely publish an xml payload to the hub.

**Pull request merged**